### PR TITLE
 LGA-1348 - Adding maintenance page

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Add maintenance page
Insert new blocks into base template to allow extending templates to customise content

## Any other changes that would benefit highlighting?

The maintenance page returns a 503 status code. This will allow angular request to display an error message if the site is put in maintenance mode whilst still on page

/status/ping.json is exempt from being redirected when in maintenance mode, if not then the loadbalancer healthchecks will fail and the instance will be taken out of routing

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_frontend/settings/base.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ def get_env_value(var_name):
 )
 
 MIDDLEWARE_CLASSES = (
+    "core.middleware.MaintenanceModeMiddleWare",
```
Comment (exonian) [inline cla_frontend/settings/base.py:121]: Convention seems to be not to capitalise the "w" in `Middleware`

File: cla_frontend/apps/core/testing/test_views.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def test_maintenance_mode_enabled_home_page(self):
+        with self.settings(MAINTENANCE_MODE=True):
+            response = self.client.get("/", follow=True)
+            self.assertEqual(503, response.status_code)
+            self.assertIn("This service is down for maintenance", response.content)
+            self.assertEqual([("http://testserver/maintenance", 302)], response.redirect_chain)
+
+    def test_maintenance_mode_enabled_maintenance_page(self):
+        with self.settings(MAINTENANCE_MODE=True):
+            response = self.client.get("/maintenance", follow=False)
```
[the remark this answers is not in the document]
Comment (said-moj) [reply cla_frontend/apps/core/testing/test_views.py:14]: Redirects.

So it's saying don't follow redirects

[the remark this answers is not in the document]
Comment (exonian) [reply cla_frontend/apps/core/testing/test_views.py:14]: Right, sorry. I meant what's the redirect here that we don't want to let it do in the test?

[the remark this answers is not in the document]
Comment (said-moj) [reply cla_frontend/apps/core/testing/test_views.py:14]: It's is probably not redirecting anything here and follow=False is being a bit over-protective.
It was intended to make sure that when `MAINTENANCE_MODE` is True then going `/maintenance` should never cause a redirect but instead should serve maintenance page with a 503 status code.

[the remark this answers is not in the document]
Comment (exonian) [reply cla_frontend/apps/core/testing/test_views.py:14]: Fair enough, that makes sense as a precaution

**Pull request merged**